portal-frontend/src/app/features/applications/edit-submission/primary-contact/primary-contact.component.ts

```diff
@@ -217,90 +203,83 @@ export class PrimaryContactComponent extends FilesStepComponent implements OnIni
+    if (!parcels || !allOwners) {
+      return;
+    }
 
-      if (selectedOwner && (this.selectedThirdPartyAgent || this.selectedLocalGovernment)) {
-        this.selectedOwnerUuid = selectedOwner.uuid;
-        this.form.patchValue({
-          firstName: selectedOwner.firstName,
-          lastName: selectedOwner.lastName,
-          organizationName: selectedOwner.organizationName,
-          phoneNumber: selectedOwner.phoneNumber,
-          email: selectedOwner.email,
-        });
-      } else if (selectedOwner) {
-        this.onSelectOwner(selectedOwner.uuid);
-      } else {
-        this.firstName.disable();
-        this.lastName.disable();
-        this.organizationName.disable();
-        this.email.disable();
-        this.phoneNumber.disable();
-      }
+    this.hasCrownParcels =
+      !!parcels && parcels.some((parcel) => parcel.ownershipTypeCode === PARCEL_OWNERSHIP_TYPE.CROWN);
+
+    const uniqueParcelOwners = parcels
+      .flatMap((parcel) => parcel.owners)
+      .reduce((map, owner) => {
+        return map.set(owner.uuid, owner);
+      }, new Map<string, ApplicationOwnerDto>());
+    const nonParcelOwners = allOwners.filter((owner) =>
+      [OWNER_TYPE.AGENT, OWNER_TYPE.GOVERNMENT].includes(owner.type.code)
+    );
+    const parcelOwners = [...uniqueParcelOwners.values()];
+    const owners = [...parcelOwners, ...nonParcelOwners];
+
+    this.parcelOwners = parcelOwners;
+    this.owners = owners;
+
+    const selectedOwner = owners.find((owner) => owner.uuid === primaryContactOwnerUuid);
+
+    // onSelectOwner only not called on first load of page
+    if (selectedOwner) {
+      this.onSelectOwner(selectedOwner.uuid);
+    } else if (parcelOwners.length === 1) {
+      this.onSelectOwner(parcelOwners[0].uuid);
+    }
 
-      if (this.isGovernmentUser || this.selectedLocalGovernment) {
-        this.prepareGovernmentOwners();
-      }
+    // Needed for L/FNG user on first load
+    this.selectedLocalGovernment = selectedOwner?.type.code === OWNER_TYPE.GOVERNMENT;
+    if (this.isGovernmentUser && this.selectedLocalGovernment) {
+      this.organizationName.setValidators([Validators.required]);
+      this.prepareGovernmentOwners();
+    } else {
+      this.organizationName.setValidators([]);
+    }
 
-      if (this.showErrors) {
-        this.form.markAllAsTouched();
-      }
-      this.isDirty = false;
-      this.calculateLetterRequired();
+    if (this.showErrors) {
+      this.form.markAllAsTouched();
+      this.ownersList.markAsTouched();
     }
+    this.isDirty = false;
+    this.calculateLetterRequired();
   }
 
   private prepareGovernmentOwners() {
     this.parcelOwners = [];
   }
 
-  async onSelectPrimaryContactType(isExistingOwner: boolean | null) {
-    if (this.form.dirty || (this.selectedOwnerUuid !== 'agent' && this.selectedOwnerUuid !== 'government')) {
+  async onSelectPrimaryContactType(event: MatButtonToggleChange) {
+    const isExistingOwner = event.value;
+
+    const isDirty =
```

Comment:
Can you use this.form.dirty to clean this up?

Reply:
That's what I had originally, but found they didn't always have the same value.

Reply:
That's because your code is checking to see if there are values in the form, not if the form is dirty. What do you actually need to check?

Reply:
Yeah, I guess what I'm really checking for is if the form is not empty. Is there a quick method for checking if all fields are empty?

Reply:
Not that I know of, I would rename it from isDirty to hasValues just to keep it clear.

Reply:
Nice cleanup!